Eastern-Research-Group/mywaterway
# Feature/arcgis upgrade 4.16

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3039786

## Main Changes:
* Upgraded the esri-loader package to 2.15.
* Upgraded the arcgis-js api to 4.16.
* Fixed highlighting issues related to changes in 4.14+
* Fixed loading spinner issues on the community page.
* Verified that random artifacts weren't being displayed on the map for point graphics. This was an issue in 4.15, but 4.16 fixed it.

## Steps To Test:
1. Try to test all map functions on the community page
2. Try to test all map functions on the state page
3. Try to test all map functions on the waterbody report page
4. Try to test all map functions on the plan summary page
5. Do a search for "Columbia, SC" on the community page
6. Verify that only 5 graphics are displayed.
  * In 4.15 there would be an extra graphic (looked like part of the stop sign) that would show up in random places.
7. Go to the state page.
8. Select South Carolina
9. Do a search on the advanced search tab
10. Verify that only 5 graphics are displayed
  * In 4.15 there would be an extra graphic (looked like part of the stop sign) that would show up in random places.

## TODO:
- [ ] Need to revisit the loading spinner on the state page. I may need to open a support ticket with esri for this one. 
  * For some reason, on the state page the "updating" flag on the view will sometimes get stuck on true. This causes an infinite loading spinner issue. This is why I used "rendering" to control the loading spinner, but "rendering" causes the loading spinner to flicker.
